slam/alarm_configuration_widget.py

```diff
@@ -231,7 +232,16 @@ def save_configuration(self):
 
         self.close()
 
-    def uncheck_enabled_box(self):
+    def uncheck_enabled_box_when_filter_set(self):
+        """
+        Grey out "Enabled" checkbox on config-page when enabling-filter is present.
+        Avoids confusion over if checkbox needs to be checked when adding filter.
+        Assume filter is valid if any text is present in text-edit (no easy way verify). 
+        """
+        editIsEmpty = not self.filter_edit.text()
+        self.enabled_checkbox.setEnabled(editIsEmpty)
+
+    def uncheck_enabled_box_when_date_set(self):
```

Comment:
This works well for typing a new filter. Also want to cover the case where a user clicks on an item that already has a filter set, it should be greyed out/disabled in that case as well. (There's probably a current bug where having a filter set sill makes it look like the alarm is enabled in the checkbox, might be similar to the last issue you are looking at).